lib/cell.rb

```diff
@@ -0,0 +1,49 @@
+class Cell
+  attr_reader :coordinate, :ship
+
+  def initialize(coordinate)
+    @coordinate = coordinate
+    @ship = nil
+    @fired_upon = false
+  end
+
+  def empty?
+    (@ship == nil)
+  end
+
+  def place_ship(ship)
+    @ship = ship
+  end
+
+  def fired_upon?
+    @fired_upon
+  end
+
+  def fire_upon
+    @fired_upon = true
+    if !(@ship == nil)
+      @ship.hit
+    end
+  end
+
+  def render(show_ships = false)
```

Comment:
As we discussed earlier, this may or may not need refactoring. Let's remember to ask during our project check-in. Thank you for taking the time to explain as we go along.